hyperon_das/query_engines.py

```diff
@@ -428,12 +428,17 @@ class RemoteQueryEngine(QueryEngine):
     def __init__(self, backend, system_parameters: Dict[str, Any], kwargs: Optional[dict] = {}):
         self.system_parameters = system_parameters
         self.local_query_engine = LocalQueryEngine(backend, kwargs)
+        self.__mode = kwargs.get('mode', 'r')
```

Comment:
I understand why you choose "r" and "w" as parameter values but I didn't like it. I prefer more verbose names like "read-only" and "read-write". 

Not just because they are more clear but because we'll probably extend this parameter to have other values which would not fit well in the analogy with file modes. For instance: "query-only"; "read-and-new-atoms" or "no-atom-changes"; "no-deletion"; etc.

Reply:
Done!